<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Brain xQTL enrichment in Glut ASoC variants</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ASoC enrichment</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">xQTL enrichment in ASoC</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Brain xQTL enrichment in Glut ASoC variants</h1>

</div>


<p>Brain xQTL(eQTL, mQTL and haQTL) genome-wide association statistics are obtained from <a href="http://mostafavilab.stat.ubc.ca/xQTLServe/">the Brain xQTL Serve</a>.</p>
<div id="enrichment-analysis-w-pi-1-statistics" class="section level2">
<h2>Enrichment analysis w/ <span class="math inline">\(\pi 1\)</span> statistics</h2>
<p>In the following <span class="math inline">\(\pi 1\)</span> analyses, for each SNP, we only keep the statistics (p value) of its nearest association.</p>
<pre class="r"><code>wd &lt;- &#39;~/Downloads/ASoC/QTL_enrich/&#39;
library(data.table)
library(dplyr)
library(ggplot2)
library(knitr)
library(kableExtra)
options(stringsAsFactors = F)
ASoC &lt;- read.delim(&#39;~/Downloads/ASoC/data/ASoC_20lines_Glut_list.txt&#39;)
pval_thres &lt;- c(6:2)
fdr_thres &lt;- c(0.01,0.05,0.1,0.25)</code></pre>
<p>We analyze the enrichment of brain xQTLs by ASoC variants by estimating <span class="math inline">\(\pi 1\)</span> of their matching p values under the xQTL test. For comparison, we generate multiple sets of ASoC variants by varying the ASoC p value/FDR threhold and estimate their <span class="math inline">\(\pi 1\)</span> levels.</p>
<pre class="r"><code># get pi1 alternative estimation
get_pi1 &lt;- function(pval,thres){
  pi1 &lt;- rep(NA,length(thres))
  for (i in 1:length(thres)){
    pi1[i] &lt;- (length(pval) - sum(pval&gt;thres[i])/(1-thres[i])) / length(pval)
  }
  return(pi1)
}
pi1_est_pval &lt;- function(brain.all.filtered, thres){
  pi1.est &lt;- data.frame(matrix(nrow = length(thres), ncol = 3))
  names(pi1.est) &lt;- c(&#39;-log10(pval) threshold&#39;,&#39;pi1 estimation&#39;,&#39;# of SNPs matched&#39;)
  for (i in 1:length(thres)){
    pi1.est[i,1] &lt;- thres[i]
    ASoC.signif.SNPs &lt;- ASoC$rsID[ASoC$X.log10P &gt; thres[i]]
    ## Get ASoC variants&#39; matching xQTL pvals:
    tmp &lt;- match(ASoC.signif.SNPs,brain.all.filtered$SNPid)
    ASoC.signif.indx &lt;- na.omit(tmp)
    # cat(length(ASoC.signif.indx),&#39;ASoC variants matched out of&#39;,length(ASoC.signif.SNPs),&#39;in total.\n&#39;)
    ASoC.pval &lt;- brain.all.filtered$pval[ASoC.signif.indx]
    pi1.est[i,2] &lt;- get_pi1(ASoC.pval,0.5)
    pi1.est[i,3] &lt;- length(ASoC.signif.indx)
    # plot1 &lt;- qplot(ASoC.pval, bins=50, xlab = &#39;ASoC QTL p values&#39;)
  }
  # pi1.control &lt;- rep(NA,5)
  # for (i in 1:5){
  #   rand.indx &lt;- sample(nrow(brain.all.filtered), 5000) ## ??? how many random SNPs
  #   ctrl.pval &lt;- brain.all.filtered$pval[rand.indx]
  #   pi1.control[i] &lt;- get_pi1(ctrl.pval,0.5)
  # } 
  print(kable(pi1.est, row.names = F, digits = 4, 
              caption = &#39;pi1 estimations at various ASoC p value significance levels:&#39;) %&gt;% 
          kable_styling(position = &#39;center&#39;,full_width = T) )
  return(pi1.est)
}
pi1_est_fdr &lt;- function(brain.all.filtered, thres){
  pi1.est &lt;- data.frame(matrix(nrow = length(thres), ncol = 3))
  names(pi1.est) &lt;- c(&#39;fdr threshold&#39;,&#39;pi1 estimation&#39;,&#39;# of SNPs matched&#39;)
  for (i in 1:length(thres)){
    pi1.est[i,1] &lt;- thres[i]
    ASoC.signif.SNPs &lt;- ASoC$rsID[ASoC$FDR &lt; thres[i]]
    ## Get ASoC variants&#39; matching xQTL pvals:
    tmp &lt;- match(ASoC.signif.SNPs,brain.all.filtered$SNPid)
    ASoC.signif.indx &lt;- na.omit(tmp)
    # cat(length(ASoC.signif.indx),&#39;ASoC variants matched out of&#39;,length(ASoC.signif.SNPs),&#39;in total.\n&#39;)
    ASoC.pval &lt;- brain.all.filtered$pval[ASoC.signif.indx]
    pi1.est[i,2] &lt;- get_pi1(ASoC.pval,0.5)
    pi1.est[i,3] &lt;- length(ASoC.signif.indx)
    # plot1 &lt;- qplot(ASoC.pval, bins=50, xlab = &#39;ASoC QTL p values&#39;)
  }
  print(kable(pi1.est, row.names = F, digits = 4, 
              caption = &#39;pi1 estimations at various ASoC fdr levels:&#39;) %&gt;% 
          kable_styling(position = &#39;center&#39;,full_width = T) )
  return(pi1.est)
}</code></pre>
<div id="brain-eqtl-enrichment" class="section level3">
<h3>Brain eQTL enrichment</h3>
<pre class="r"><code>brain.QTLs_by_dist &lt;- as.data.frame(fread(paste0(wd,&#39;Brain_eQTLs_by_dist.w_zscores.txt&#39;)))
eQTL.pi1_pval &lt;- pi1_est_pval(brain.QTLs_by_dist, pval_thres)</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
pi1 estimations at various ASoC p value significance levels:
</caption>
<thead>
<tr>
<th style="text-align:right;">
-log10(pval) threshold
</th>
<th style="text-align:right;">
pi1 estimation
</th>
<th style="text-align:right;">
# of SNPs matched
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.4051
</td>
<td style="text-align:right;">
1012
</td>
</tr>
<tr>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.4051
</td>
<td style="text-align:right;">
1402
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.3770
</td>
<td style="text-align:right;">
2048
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0.3687
</td>
<td style="text-align:right;">
3352
</td>
</tr>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.3414
</td>
<td style="text-align:right;">
6453
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>eQTL.pi1_fdr &lt;- pi1_est_fdr(brain.QTLs_by_dist, fdr_thres)</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
pi1 estimations at various ASoC fdr levels:
</caption>
<thead>
<tr>
<th style="text-align:right;">
fdr threshold
</th>
<th style="text-align:right;">
pi1 estimation
</th>
<th style="text-align:right;">
# of SNPs matched
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.3810
</td>
<td style="text-align:right;">
2575
</td>
</tr>
<tr>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.3608
</td>
<td style="text-align:right;">
4199
</td>
</tr>
<tr>
<td style="text-align:right;">
0.10
</td>
<td style="text-align:right;">
0.3412
</td>
<td style="text-align:right;">
5665
</td>
</tr>
<tr>
<td style="text-align:right;">
0.25
</td>
<td style="text-align:right;">
0.3533
</td>
<td style="text-align:right;">
10208
</td>
</tr>
</tbody>
</table>
<div id="negative-control-using-random-snps-in-ocrs" class="section level4">
<h4>Negative control using random SNPs in OCRs</h4>
<p>For negative control, we look at the p value distribution of random QTLs that fall into the open chromatin regions (OCRs) of Glutamatergic cell lines (peaks called from all 20 Glut lines) and estimate its <span class="math inline">\(\pi 1\)</span>.</p>
<pre class="r"><code>QTLs_in_OCR.df &lt;- as.data.frame(fread(paste0(wd,&#39;QTL_in_OCR/brain_eQTL_hg38_in_CN_peaks.bed&#39;)))
names(QTLs_in_OCR.df) &lt;- c(&#39;chr&#39;,&#39;start&#39;,&#39;end&#39;,&#39;rsID&#39;)
QTLs_in_OCR.pval &lt;- brain.QTLs_by_dist$pval[match(QTLs_in_OCR.df$rsID, brain.QTLs_by_dist$SNPid)]
QTL.random.pi1 &lt;- get_pi1(QTLs_in_OCR.pval,0.5)
## Get a distribution
# random.pi1 &lt;- rep(NA,1000)
# for (i in 1:1000){
#   rand.indx &lt;- sample(length(eQTLs_in_OCR.pval),5000)
#   random.pi1[i] &lt;- get_pi1(eQTLs_in_OCR.pval[rand.indx],0.5)
# }
eQTL.pi1_pval[length(pval_thres)+1,1] &lt;- &#39;random&#39;
eQTL.pi1_pval[length(pval_thres)+1,2] &lt;- QTL.random.pi1
eQTL.pi1_fdr[length(fdr_thres)+1,1] &lt;- &#39;random&#39;
eQTL.pi1_fdr[length(fdr_thres)+1,2] &lt;- QTL.random.pi1
cat(&#39;pi1 of random control:&#39;, QTL.random.pi1)</code></pre>
<pre><code>pi1 of random control: 0.2708471</code></pre>
</div>
<div id="positive-control-using-asoc-variants-in-adult-brain-ocrs" class="section level4">
<h4>Positive control using ASoC variants in adult brain OCRs</h4>
<p>For positive control, we look at the QTL p value distribution of ASoC variants (FDR &lt; 0.05) that fall into the adult brain OCRs (PsychENCODE) and estimate the <span class="math inline">\(\pi 1\)</span>.</p>
<pre class="r"><code>ASoC.adult_OCR &lt;- as.data.frame(fread(paste0(wd,&#39;QTL_in_OCR/ASoC_Glut20_in_adult_OCR.bed&#39;)))
names(ASoC.adult_OCR) &lt;- c(&#39;chr&#39;,&#39;start&#39;,&#39;end&#39;,&#39;rsID&#39;,&#39;FDR&#39;)
ASoC.adult_OCR.signif &lt;- filter(ASoC.adult_OCR, FDR&lt;0.05)
cat(&#39;Out of 5611 ASoC variants,&#39;,nrow(ASoC.adult_OCR.signif),&#39;fall in adult brain OCRs.&#39;)</code></pre>
<pre><code>Out of 5611 ASoC variants, 1665 fall in adult brain OCRs.</code></pre>
<pre class="r"><code>match.indx &lt;- match(ASoC.adult_OCR.signif$rsID,brain.QTLs_by_dist$SNPid)
match.indx &lt;- na.omit(match.indx)
QTL.adult.pi1 &lt;- get_pi1(brain.QTLs_by_dist$pval[match.indx],0.5) # 0.4551551

eQTL.pi1_pval[length(pval_thres)+2,1] &lt;- &#39;positive&#39;
eQTL.pi1_pval[length(pval_thres)+2,2] &lt;- QTL.adult.pi1
eQTL.pi1_fdr[length(fdr_thres)+2,1] &lt;- &#39;positive&#39;
eQTL.pi1_fdr[length(fdr_thres)+2,2] &lt;- QTL.adult.pi1
cat(&#39;pi1 of&#39;, length(match.indx),&#39;matched positive control SNPs:&#39;, QTL.adult.pi1)</code></pre>
<pre><code>pi1 of 1193 matched positive control SNPs: 0.4551551</code></pre>
</div>
</div>
<div id="brain-mqtl-enrichment" class="section level3">
<h3>Brain mQTL enrichment</h3>
<pre class="r"><code>brain.QTLs_by_dist &lt;- as.data.frame(fread(paste0(wd,&#39;Brain_mQTLs_by_dist.w_zscores.txt&#39;)))
mQTL.pi1_pval &lt;- pi1_est_pval(brain.QTLs_by_dist, pval_thres)</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
pi1 estimations at various ASoC p value significance levels:
</caption>
<thead>
<tr>
<th style="text-align:right;">
-log10(pval) threshold
</th>
<th style="text-align:right;">
pi1 estimation
</th>
<th style="text-align:right;">
# of SNPs matched
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.4727
</td>
<td style="text-align:right;">
1009
</td>
</tr>
<tr>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.4835
</td>
<td style="text-align:right;">
1390
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.4733
</td>
<td style="text-align:right;">
2039
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0.4462
</td>
<td style="text-align:right;">
3344
</td>
</tr>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.4309
</td>
<td style="text-align:right;">
6463
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>mQTL.pi1_fdr &lt;- pi1_est_fdr(brain.QTLs_by_dist, fdr_thres)</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
pi1 estimations at various ASoC fdr levels:
</caption>
<thead>
<tr>
<th style="text-align:right;">
fdr threshold
</th>
<th style="text-align:right;">
pi1 estimation
</th>
<th style="text-align:right;">
# of SNPs matched
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.4565
</td>
<td style="text-align:right;">
2565
</td>
</tr>
<tr>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.4367
</td>
<td style="text-align:right;">
4200
</td>
</tr>
<tr>
<td style="text-align:right;">
0.10
</td>
<td style="text-align:right;">
0.4335
</td>
<td style="text-align:right;">
5680
</td>
</tr>
<tr>
<td style="text-align:right;">
0.25
</td>
<td style="text-align:right;">
0.4253
</td>
<td style="text-align:right;">
10249
</td>
</tr>
</tbody>
</table>
<div id="negative-control-using-random-snps-in-ocrs-1" class="section level4">
<h4>Negative control using random SNPs in OCRs</h4>
<pre class="r"><code>QTLs_in_OCR.df &lt;- as.data.frame(fread(paste0(wd,&#39;QTL_in_OCR/brain_mQTL_hg38_in_CN_peaks.bed&#39;)))
names(QTLs_in_OCR.df) &lt;- c(&#39;chr&#39;,&#39;start&#39;,&#39;end&#39;,&#39;rsID&#39;)
QTLs_in_OCR.pval &lt;- brain.QTLs_by_dist$pval[match(QTLs_in_OCR.df$rsID, brain.QTLs_by_dist$SNPid)]
QTL.random.pi1 &lt;- get_pi1(QTLs_in_OCR.pval,0.5)

mQTL.pi1_pval[length(pval_thres)+1,1] &lt;- &#39;random&#39;
mQTL.pi1_pval[length(pval_thres)+1,2] &lt;- QTL.random.pi1
mQTL.pi1_fdr[length(fdr_thres)+1,1] &lt;- &#39;random&#39;
mQTL.pi1_fdr[length(fdr_thres)+1,2] &lt;- QTL.random.pi1
cat(&#39;pi1 of random control:&#39;, QTL.random.pi1)</code></pre>
<pre><code>pi1 of random control: 0.35418</code></pre>
</div>
<div id="positive-control-using-asoc-variants-in-adult-brain-ocrs-1" class="section level4">
<h4>Positive control using ASoC variants in adult brain OCRs</h4>
<pre class="r"><code>match.indx &lt;- match(ASoC.adult_OCR.signif$rsID,brain.QTLs_by_dist$SNPid)
match.indx &lt;- na.omit(match.indx)
QTL.adult.pi1 &lt;- get_pi1(brain.QTLs_by_dist$pval[match.indx],0.5) # 

mQTL.pi1_pval[length(pval_thres)+2,1] &lt;- &#39;positive&#39;
mQTL.pi1_pval[length(pval_thres)+2,2] &lt;- QTL.adult.pi1
mQTL.pi1_fdr[length(fdr_thres)+2,1] &lt;- &#39;positive&#39;
mQTL.pi1_fdr[length(fdr_thres)+2,2] &lt;- QTL.adult.pi1
cat(&#39;pi1 of&#39;, length(match.indx),&#39;matched positive control SNPs:&#39;, QTL.adult.pi1)</code></pre>
<pre><code>pi1 of 1198 matched positive control SNPs: 0.4741235</code></pre>
</div>
</div>
<div id="brain-haqtl-enrichment" class="section level3">
<h3>Brain haQTL enrichment</h3>
<pre class="r"><code>brain.QTLs_by_dist &lt;- as.data.frame(fread(paste0(wd,&#39;Brain_haQTLs_by_dist.w_zscores.txt&#39;)))
haQTL.pi1_pval &lt;- pi1_est_pval(brain.QTLs_by_dist, pval_thres)</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
pi1 estimations at various ASoC p value significance levels:
</caption>
<thead>
<tr>
<th style="text-align:right;">
-log10(pval) threshold
</th>
<th style="text-align:right;">
pi1 estimation
</th>
<th style="text-align:right;">
# of SNPs matched
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.3824
</td>
<td style="text-align:right;">
1033
</td>
</tr>
<tr>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.4000
</td>
<td style="text-align:right;">
1430
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.3694
</td>
<td style="text-align:right;">
2090
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0.3516
</td>
<td style="text-align:right;">
3430
</td>
</tr>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.3404
</td>
<td style="text-align:right;">
6592
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>haQTL.pi1_fdr &lt;- pi1_est_fdr(brain.QTLs_by_dist, fdr_thres)</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
pi1 estimations at various ASoC fdr levels:
</caption>
<thead>
<tr>
<th style="text-align:right;">
fdr threshold
</th>
<th style="text-align:right;">
pi1 estimation
</th>
<th style="text-align:right;">
# of SNPs matched
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.3625
</td>
<td style="text-align:right;">
2629
</td>
</tr>
<tr>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.3450
</td>
<td style="text-align:right;">
4293
</td>
</tr>
<tr>
<td style="text-align:right;">
0.10
</td>
<td style="text-align:right;">
0.3407
</td>
<td style="text-align:right;">
5791
</td>
</tr>
<tr>
<td style="text-align:right;">
0.25
</td>
<td style="text-align:right;">
0.3220
</td>
<td style="text-align:right;">
10448
</td>
</tr>
</tbody>
</table>
<div id="negative-control-using-random-snps-in-ocrs-2" class="section level4">
<h4>Negative control using random SNPs in OCRs</h4>
<pre class="r"><code>QTLs_in_OCR.df &lt;- as.data.frame(fread(paste0(wd,&#39;QTL_in_OCR/brain_haQTL_hg38_in_CN_peaks.bed&#39;)))
names(QTLs_in_OCR.df) &lt;- c(&#39;chr&#39;,&#39;start&#39;,&#39;end&#39;,&#39;rsID&#39;)
QTLs_in_OCR.pval &lt;- brain.QTLs_by_dist$pval[match(QTLs_in_OCR.df$rsID, brain.QTLs_by_dist$SNPid)]
QTL.random.pi1 &lt;- get_pi1(QTLs_in_OCR.pval,0.5)

haQTL.pi1_pval[length(pval_thres)+1,1] &lt;- &#39;random&#39;
haQTL.pi1_pval[length(pval_thres)+1,2] &lt;- QTL.random.pi1
haQTL.pi1_fdr[length(fdr_thres)+1,1] &lt;- &#39;random&#39;
haQTL.pi1_fdr[length(fdr_thres)+1,2] &lt;- QTL.random.pi1
cat(&#39;pi1 of random control:&#39;, QTL.random.pi1)</code></pre>
<pre><code>pi1 of random control: 0.2131081</code></pre>
</div>
<div id="positive-control-using-asoc-variants-in-adult-brain-ocrs-2" class="section level4">
<h4>Positive control using ASoC variants in adult brain OCRs</h4>
<pre class="r"><code>match.indx &lt;- match(ASoC.adult_OCR.signif$rsID,brain.QTLs_by_dist$SNPid)
match.indx &lt;- na.omit(match.indx)
QTL.adult.pi1 &lt;- get_pi1(brain.QTLs_by_dist$pval[match.indx],0.5) # 

haQTL.pi1_pval[length(pval_thres)+2,1] &lt;- &#39;positive&#39;
haQTL.pi1_pval[length(pval_thres)+2,2] &lt;- QTL.adult.pi1
haQTL.pi1_fdr[length(fdr_thres)+2,1] &lt;- &#39;positive&#39;
haQTL.pi1_fdr[length(fdr_thres)+2,2] &lt;- QTL.adult.pi1
cat(&#39;pi1 of&#39;, length(match.indx),&#39;matched positive control SNPs:&#39;, QTL.adult.pi1)</code></pre>
<pre><code>pi1 of 1204 matched positive control SNPs: 0.4551495</code></pre>
</div>
</div>
</div>
<div id="summary-of-pi-1-enrichment-results" class="section level2">
<h2>Summary of <span class="math inline">\(\pi 1\)</span> enrichment results</h2>
<div id="varing-asoc-p-value-threshold" class="section level3">
<h3>Varing ASoC p value threshold</h3>
<pre class="r"><code>pi1_pval &lt;- rbind(eQTL.pi1_pval,mQTL.pi1_pval,haQTL.pi1_pval)
names(pi1_pval) &lt;- c(&#39;thres&#39;,&#39;pi1&#39;,&#39;num_SNPs&#39;)
pi1_pval$type &lt;- rep(c(&#39;eQTL&#39;,&#39;mQTL&#39;,&#39;haQTL&#39;), each=nrow(eQTL.pi1_pval))
pi1_pval$thres &lt;- factor(pi1_pval$thres)
pi1_pval$thres &lt;- relevel(pi1_pval$thres,&#39;random&#39;)
ggplot(pi1_pval, aes(x=factor(type), y=pi1, fill = thres)) + 
  geom_bar(stat=&quot;identity&quot;, position = &#39;dodge&#39;) + theme_minimal() + # ylim(low=0,high=0.6) +
  theme(axis.title.x = element_blank(), legend.position=&quot;bottom&quot;) + 
  labs(fill = &quot;-log10(p value) cut-off&quot;) + ylim(0,0.5)</code></pre>
<p><img src="Figs/combined%20pval-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="varing-asoc-fdr-threshold" class="section level3">
<h3>Varing ASoC FDR threshold</h3>
<pre class="r"><code>pi1_fdr &lt;- rbind(eQTL.pi1_fdr,mQTL.pi1_fdr,haQTL.pi1_fdr)
names(pi1_fdr) &lt;- c(&#39;thres&#39;,&#39;pi1&#39;,&#39;num_SNPs&#39;)
pi1_fdr$type &lt;- rep(c(&#39;eQTL&#39;,&#39;mQTL&#39;,&#39;haQTL&#39;), each=nrow(eQTL.pi1_fdr))
pi1_fdr$thres &lt;- factor(pi1_fdr$thres, levels = c(&#39;random&#39;,&#39;0.25&#39;,&#39;0.1&#39;,&#39;0.05&#39;,&#39;0.01&#39;,&#39;positive&#39;))
ggplot(pi1_fdr, aes(x=factor(type), y=pi1, fill = thres)) + 
  geom_bar(stat=&quot;identity&quot;, position = &#39;dodge&#39;) + theme_minimal() + # ylim(low=0,high=0.6) +
  theme(axis.title.x = element_blank(), legend.position=&quot;bottom&quot;) + 
  labs(fill = &quot;FDR cut-off&quot;) + ylim(0,0.5)</code></pre>
<p><img src="Figs/combined%20fdr-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="torus-enrichment-analysis" class="section level2">
<h2>TORUS enrichment analysis</h2>
<pre class="r"><code>torus.df &lt;- read.delim(paste0(wd,&#39;torus_result_table.txt&#39;), header = T,row.names = 1)
kable(torus.df) %&gt;% kable_styling(position = &#39;center&#39;,full_width = T)</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
log.oddsRatio.
</th>
<th style="text-align:right;">
low.CI
</th>
<th style="text-align:right;">
high.CI
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
eQTL_all
</td>
<td style="text-align:right;">
3.474
</td>
<td style="text-align:right;">
2.260
</td>
<td style="text-align:right;">
4.688
</td>
</tr>
<tr>
<td style="text-align:left;">
mQTL_all
</td>
<td style="text-align:right;">
3.561
</td>
<td style="text-align:right;">
3.006
</td>
<td style="text-align:right;">
4.116
</td>
</tr>
<tr>
<td style="text-align:left;">
haQTL_all
</td>
<td style="text-align:right;">
4.477
</td>
<td style="text-align:right;">
3.820
</td>
<td style="text-align:right;">
5.133
</td>
</tr>
<tr>
<td style="text-align:left;">
neg_eQTL
</td>
<td style="text-align:right;">
1.176
</td>
<td style="text-align:right;">
0.831
</td>
<td style="text-align:right;">
1.522
</td>
</tr>
<tr>
<td style="text-align:left;">
neg_mQTL
</td>
<td style="text-align:right;">
1.434
</td>
<td style="text-align:right;">
1.246
</td>
<td style="text-align:right;">
1.622
</td>
</tr>
<tr>
<td style="text-align:left;">
pos_eQTL
</td>
<td style="text-align:right;">
1.746
</td>
<td style="text-align:right;">
-21.439
</td>
<td style="text-align:right;">
24.932
</td>
</tr>
<tr>
<td style="text-align:left;">
pos_mQTL
</td>
<td style="text-align:right;">
4.141
</td>
<td style="text-align:right;">
-0.409
</td>
<td style="text-align:right;">
8.690
</td>
</tr>
</tbody>
</table>
<p>We use TORUS to estimate the enrichment of Glut ASoC variants for xQTL SNPs. The table presents the log2-odds ratios and their 95% confidence intervals.</p>
<p>Unlike the previous <span class="math inline">\(\pi 1\)</span> analyses, we keep all of the QTL statistics per SNP.</p>
<p>For negative control, we use ATAC-seq peak region obtained from 20 Glut cell lines as annotation.</p>
<p>For positive control, we use ASoC variants (FDR &lt; 0.05) that fall into the adult brain OCRs (PsychENCODE) as annotation. Since the number of ASoC variants that overlap with xQTL is small, the confidence intervals are very wide.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
